storage/integration_test.go

```diff
@@ -3428,199 +3428,195 @@ func TestIntegration_CancelWrite(t *testing.T) {
+						MaxAge:          time.Hour,
+						Methods:         []string{"GET"},
+						Origins:         []string{"*"},
+						ResponseHeaders: []string{"some-header"},
+					},
 				},
-			},
-			want: []CORS{
-				{
-					MaxAge:          time.Hour,
-					Methods:         []string{"GET"},
-					Origins:         []string{"*"},
-					ResponseHeaders: []string{"some-header"},
+				want: []CORS{
+					{
+						MaxAge:          time.Hour,
+						Methods:         []string{"GET"},
+						Origins:         []string{"*"},
+						ResponseHeaders: []string{"some-header"},
+					},
 				},
 			},
-		},
-		{
-			input: []CORS{},
-			want:  nil,
-		},
-		{
-			input: nil,
-			want: []CORS{
-				{
-					MaxAge:          time.Hour,
-					Methods:         []string{"POST"},
-					Origins:         []string{"some-origin.com"},
-					ResponseHeaders: []string{"foo-bar"},
+			{
+				desc:  "set to empty to remove existing policies",
+				input: []CORS{},
+				want:  nil,
+			},
+			{
+				desc:  "do not set to keep existing policies",
+				input: nil,
+				want: []CORS{
+					{
+						MaxAge:          time.Hour,
+						Methods:         []string{"POST"},
+						Origins:         []string{"some-origin.com"},
+						ResponseHeaders: []string{"foo-bar"},
+					},
 				},
 			},
-		},
-	} {
-		bkt := client.Bucket(uidSpace.New())
-		h.mustCreate(bkt, testutil.ProjID(), &BucketAttrs{CORS: initialSettings})
-		defer h.mustDeleteBucket(bkt)
-		h.mustUpdateBucket(bkt, BucketAttrsToUpdate{CORS: test.input}, h.mustBucketAttrs(bkt).MetaGeneration)
-		attrs := h.mustBucketAttrs(bkt)
-		if diff := testutil.Diff(attrs.CORS, test.want); diff != "" {
-			t.Errorf("input: %v\ngot=-, want=+:\n%s", test.input, diff)
+		} {
+			t.Run(test.desc, func(t *testing.T) {
+				h := testHelper{t}
+
+				bkt := client.Bucket(prefix + uidSpace.New())
+				h.mustCreate(bkt, testutil.ProjID(), &BucketAttrs{CORS: initialSettings})
+				defer h.mustDeleteBucket(bkt)
+				// Set VersioningEnabled attr so we don't fail when setting CORS to nil
+				h.mustUpdateBucket(bkt, BucketAttrsToUpdate{CORS: test.input, VersioningEnabled: false}, h.mustBucketAttrs(bkt).MetaGeneration)
+				attrs := h.mustBucketAttrs(bkt)
+				if diff := testutil.Diff(attrs.CORS, test.want); diff != "" {
+					t.Errorf("input: %v\ngot=-, want=+:\n%s", test.input, diff)
+				}
+			})
 		}
-	}
+	})
 }
 
 func TestIntegration_UpdateDefaultEventBasedHold(t *testing.T) {
-	ctx := context.Background()
-	client := testConfig(ctx, t)
-	defer client.Close()
-	h := testHelper{t}
+	multiTransportTest(context.Background(), t, func(t *testing.T, ctx context.Context, _ string, prefix string, client *Client) {
+		h := testHelper{t}
 
-	bkt := client.Bucket(uidSpace.New())
-	h.mustCreate(bkt, testutil.ProjID(), &BucketAttrs{})
-	defer h.mustDeleteBucket(bkt)
-	attrs := h.mustBucketAttrs(bkt)
-	if attrs.DefaultEventBasedHold != false {
-		t.Errorf("got=%v, want=%v", attrs.DefaultEventBasedHold, false)
-	}
+		bkt := client.Bucket(prefix + uidSpace.New())
```

Comment:
Do we need to use prefix here if we are only using one bucket?

Reply:
This allows a diff prefix between http and grpc and is needed right now for grpc.

Reply:
While the allowlist is in place, we should always create our test buckets with the prefix. It is empty for `http` tests fwiw.

Reply:
I was asking why we are using the `prefix` arg to `multiTransportTest` rather than the `bucket` arg. But I think I was confused anyway, we need to create a bucket for this test rather than using the shared one because we want to be modifying the CORS config.